modules/ProjectScore.py

```diff
@@ -0,0 +1,186 @@
+"""Disease to target evidence parser for Project Score v2."""
+import argparse
+import logging
+
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql import functions as f
+from pyspark.sql import types as t
+
+from common.evidence import (
+    GenerateDiseaseCellLines,
+    initialize_logger,
+    initialize_sparksession,
+    write_evidence_strings,
+)
+
+# This parser is specific for the second release of Porject Score, so the publication identifier is hardcoded
+PMID = "38215750"
+
+
+def generate_project_score_evidence(
+    project_score_cell_lines: DataFrame, project_score_evidence: DataFrame
+) -> DataFrame:
+    """Generate evidence strings for Project Score v2.
+
+    Args:
+        project_score_cell_lines (DataFrame): DataFrame containing cell line annotations
+        project_score_evidence (DataFrame): DataFrame containing evidence data
+
+    Returns:
+        DataFrame: DataFrame containing evidence strings
+    """
+    pmid = "38215750"
+    return (
+        project_score_evidence.select(
+            f.col("targetSymbol").alias("targetFromSource"),
+            f.col("diseaseName").alias("diseaseFromSource"),
+            f.col("diseaseId").alias("diseaseFromSourceMappedId"),
+            f.col("PRIORITY").cast(t.FloatType()).alias("resourceScore"),
+            f.col("targetId").alias("targetFromSourceId"),
+            f.array(f.lit(pmid)).alias("literature"),
+            f.lit("crispr").alias("datasourceId"),
+            f.lit("affected_pathway").alias("datatypeId"),
+            f.lower(f.col("cancerType")).alias("cancerType"),
+        )
+        .join(project_score_cell_lines, on="cancerType", how="left")
+        # Dropping the pancancer genes:
+        .filter(f.col("cancerType") != "pancancer")
+        # Cleaning table:
+        .drop("cancerType")
+    )
+
+
+def main(
+    spark: SparkSession,
+    evid_file: str,
+    cell_line_file: str,
+    cell_passport_file: str,
+    cell_line_to_uberon_mapping: str,
+    out_file: str,
+) -> None:
+    """Main function for parsing Project Score v2 data."""
+    # Get logger:
+    logger = logging.getLogger(__name__)
+
+    # Logging command parameters:
+    logger.info("Running Project Score parser with parameters:")
+    logger.info(f"Evidence file: {evid_file}")
+    logger.info(f"Cell types file: {cell_line_file}")
+    logger.info(f"Cell passport file: {cell_passport_file}")
+    logger.info(f"Cell line to uberon mapping: {cell_line_to_uberon_mapping}")
+    logger.info(f"Output file: {out_file}")
+
+    # Extract disease cell line data from cell passport file:
+    cell_passport_data = GenerateDiseaseCellLines(
+        spark, cell_passport_file, cell_line_to_uberon_mapping
+    )
+
+    passport_disease_cell_lines = cell_passport_data.generate_disease_cell_lines()
+
+    # Joining disease cell-lines dataframe for Project Score:
+    disease_cell_lines = (
```

Comment:
I'd suggest to extract this logic into a function

Reply:
Interesting idea. I moved all cell-line related logic into a separate function things become neater. Thanks.

Reply:
Thanks!